src/query/management/src/role/role_mgr.rs

```diff
@@ -256,32 +258,43 @@ impl RoleApi for RoleMgr {
     #[minitrace::trace]
     async fn transfer_ownership_to_admin(
         &self,
-        object: &OwnershipObject,
+        objects: &Vec<OwnershipObject>,
     ) -> databend_common_exception::Result<()> {
         // Ensure accurate matching of a key
-        let match_seq = MatchSeq::GE(1);
-        let key = self.ownership_object_key(object);
+        let mut if_then = vec![];
+        for object in objects {
+            let owner_key = self.ownership_object_key(object);
+            let owner_value = serialize_struct(
+                &OwnershipInfo {
+                    object: object.clone(),
+                    role: "account_admin".to_string(),
+                },
+                ErrorCode::IllegalUserInfoFormat,
+                || "",
+            )?;
+            if_then.push(txn_op_put(&owner_key, owner_value))
+        }
 
-        let value = serialize_struct(
-            &OwnershipInfo {
-                object: object.clone(),
-                role: "account_admin".to_string(),
-            },
-            ErrorCode::IllegalUserInfoFormat,
-            || "",
-        )?;
+        let mut retry = 0;
 
-        let kv_api = self.kv_api.clone();
-        kv_api
-            .upsert_kv(UpsertKVReq::new(
-                &key,
-                match_seq,
-                Operation::Update(value),
-                None,
-            ))
-            .await?;
+        let txn_req = TxnRequest {
+            condition: vec![],
+            if_then: if_then.clone(),
+            else_then: vec![],
+        };
 
-        Ok(())
+        while retry < TXN_MAX_RETRY_TIMES {
```

Comment:
Use `txn_backoff()` to implement the while-loop:
https://github.com/datafuselabs/databend/blob/dac261a319f80dc37ef3ace7cde21428ef21c5a6/src/meta/api/src/schema_api_impl.rs#L1568